### PR TITLE
feat: Add huggingface support

### PR DESCRIPTION
**Pull request by Xuanwo**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

Allowing loading date from huggingface!

```sql
MySQL [(none)]> SELECT *
    -> FROM 'hf://opendal/huggingface-testdata/default/train/' (
    ->     connection => (
    ->         token = 'xxxxxxxxxxxxxxxxxxxxxxx',
    ->         revision = 'refs%2Fconvert%2Fparquet'
    ->     ),
    ->     PATTERN => '.*parquet'
    -> )
    -> LIMIT 20;
```

## Tests

- [x] Unit Test
- [x] Logic Test
- [x] Benchmark Test
- [ ] No Test  - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/14304)

### CODE REVIEW

**Comment by Xuanwo** [comment]
cc @soyeric128, we added a new storage type for stage: huggingface.

The URI format: `hf://{repo_id}/path/to/file`, `repo_id` looks like `opendal/huggingface-testdata`.

Supported config includes:

- `repo_type`: the repo type for your huggingface uri, default to `dataset`, available `dataset`, `model`
- `revision`: the revision for yor huggingface uri, default to `main`. Could be a branch, tag, or commit of your repo.
- `token`: the API token from `huggingface`.

**Comment by BohuTANG** [comment]
Does the `hf` can be accessed anonymously?

**Comment by Xuanwo** [comment]
> Does the `hf` can be accessed anonymously? Say:
> 
> ```
> SELECT *
>     -> FROM 'hf://opendal/huggingface-testdata/default/train/' (
>     ->     PATTERN => '.*parquet'
>     -> )
>     -> LIMIT 20;
> ```

Yes!

image

**Review by drmingdrmer — APPROVED** [review]
> Reviewed 9 of 11 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/datafuselabs/databend/14304)* status: 9 of 11 files reviewed, all discussions resolved (waiting on @BohuTANG)

File: src/meta/app/src/storage/storage_params.rs [hunk 1]
```diff
@@ -562,6 +572,29 @@ impl Debug for StorageCosConfig {
     }
 }
 
+#[derive(Clone, Default, PartialEq, Eq, Serialize, Deserialize)]
+pub struct StorageHuggingfaceConfig {
+    pub repo_id: String,
+    pub repo_type: String,
+    pub revision: String,
```
Comment (BohuTANG) [inline src/meta/app/src/storage/storage_params.rs:579]: Add more comments for the fields?

Comment (Xuanwo) [reply src/meta/app/src/storage/storage_params.rs:588]: Addressed.

**Pull request merged**